src/backends/interflop-mca-int/interflop_mca_int.c

```diff
@@ -201,28 +195,22 @@ static void _noise_binary128(__float128 *x, const int exp,
                              rng_state_t *rng_state) {
 
   // Convert preserving-bytes __float128 to __int128
-  __int128 x_s128 = *(__int128 *)(x);
+  binary128 *b128 = (binary128 *)x;
 
   // amount by which to shift the noise term sign (1) + exp (15) + noise
   // exponent
-  uint32_t shift = 1 + QUAD_EXP_SIZE - exp;
+  const uint32_t shift = 1 + QUAD_EXP_SIZE - exp;
 
   // Generate 128 signed noise
   // only 64 bits of noise are used, they are left aligned in a signed 64 bit
-  int64_t noise_high =
-      _get_rand_uint64(rng_state, &global_tid_lock, &global_tid);
-  __int128 noise = noise_high;
-  noise <<= 64;
+  binary128 noise = {.words64.high = get_rand_uint64(rng_state, &global_tid)};
 
   // right shift the noise to the correct magnitude, this is a arithmetic shift
   // and sign bit will be extended
-  noise = noise >> shift;
+  noise.i128 >>= shift;
 
   // Add the noise
-  x_s128 += noise;
-
-  // Convert back to __float128
-  *x = *(__float128 *)(&x_s128);
+  b128->i128 += noise.i128;
```

Comment:
Are you sure the masks are not more costly ? I find the previous cast pretty clean here. But if you prefer the unions, I'm ok.

Reply:
Yes it is less costly and it's actually where we save cycles.
With the last cast, the compiler generates `mov` instructions.
![image](https://user-images.githubusercontent.com/7467329/178527589-c9e16d57-5182-404a-9d58-da75af5d58cd.png)

Using the union, we generate `add` instructions, which is faster.
![image](https://user-images.githubusercontent.com/7467329/178527297-b633fe94-7c38-4619-be7e-355c02a1d5b6.png)

Reply:
Oh wow, I would have never guessed :clap: